sources/Application/Views/ModalDialogs/PagedImportSampleDialog.cpp

```diff
@@ -168,19 +158,37 @@ void PagedImportSampleDialog::ProcessButtonMask(unsigned short mask,
 
   if (!pressed)
     return;
-
-  if (mask & EPBM_B) {
+  // make sure to index into the fileList with the offset from topIndex!
+  FileListItem currentItem = fileList_[currentSample_ - topIndex_];
+
+  auto fullPathStr = std::string(currentPath_.GetPath());
+  fullPathStr += "/";
+  fullPathStr += currentItem.name;
+  auto fullPath = Path{fullPathStr};
+
+  if (mask & EPBM_START) {
+    if (mask & EPBM_R) {
```

Comment:
Cosmetic, but wouldn't it be more clear if EPBM_R is in the outer if here? given that you'd do "shift+key" rather than "key+shift"?

Reply:
actually I think I'll keep this as is, as the outer if handles both preview and import, if I swap them around, I'll then need an extra if statement to check for start (play) on its own.